### PR TITLE
added controller, dto, mapper, exception and config packages

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> findAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping ("/{id}")
+    public BookDto findBookById(@PathVariable Long id) {
+        return bookService.findById(id);
```
Comment (forzen123212) [inline src/main/java/mate/academy/bookshop/controller/BookController.java:28]: Краще переіменувати findBookById на getById 
find ми використовуєм коли повертаємо Optional<BookDto> 
get коли точне значення яке не може бути null

File: src/main/java/mate/academy/bookshop/service/BookServiceImpl.java [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("" + new Random().nextInt(1000));
+        return bookMapper.toDto(bookRepository.save(book));
```
Comment (forzen123212) [inline src/main/java/mate/academy/bookshop/service/BookServiceImpl.java]: implement isbm initialization into mapper

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,8 +1,11 @@
-spring.datasource.url=jdbc:mysql://localhost:3306/ticket_app?serverTimeZone=UTC
+spring.datasource.url=jdbc:mysql://localhost:3306/book_shop
 spring.datasource.username=root
-spring.datasource.password=x
+spring.datasource.password=fiasko13@3
```
Comment (forzen123212) [inline src/main/resources/application.properties]: remove your pass

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.MySQLDialect
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+
```
Comment (forzen123212) [inline src/main/resources/application.properties]: remove empty line

**Review by forzen123212 — APPROVED** [review]
good job, just some style fixes!

File: src/main/java/mate/academy/bookshop/controller/BookController.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/bookshop/controller/BookController.java:28]: It's a good suggestion, but not required.

File: src/main/java/mate/academy/bookshop/service/BookServiceImpl.java [hunk 5]
```diff
@@ -1,27 +1,41 @@
...
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("" + new Random().nextInt(1000));
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/service/BookServiceImpl.java]: Why is it generated? I'm not sure that stores generate ISBN of some book they want to sell. So, make it an API parameter.

File: src/main/resources/application.properties (hunk 3 above)
Comment (vovasalyha) [reply src/main/resources/application.properties]: Yep, better use some dummy passwords.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: src/main/java/mate/academy/bookshop/service/BookServiceImpl.java (hunk 5 above)
Comment (BolsunovaNataliia) [reply src/main/java/mate/academy/bookshop/service/BookServiceImpl.java]: :) A little drifted.

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**